### PR TITLE
osd/scrub: remove the 'penalty queue' from the scrubber

### PR DESCRIPTION
**Pull request by ronen-fr**
One more step towards a scrub scheduler that implements
the design as presented in 2023. Changes here:
- introducing the 'not before' delay mechanism. For now -
  the nb attribute is attached to the whole scrub job
  (which is not separated into shallow & deep targets yet).
- removing the 'penalty queue' from the code. Using the
  'not before' delay mechanism instead.

### CODE REVIEW

File: src/osd/scrubber/osd_scrub.cc [hunk 1]
```diff
@@ -429,11 +429,22 @@ Scrub::sched_params_t OsdScrub::determine_scrub_time(
...
     Scrub::ScrubJobRef sjob,
-    const Scrub::sched_params_t& suggested)
+    const Scrub::sched_params_t& suggested,
+    bool reset_notbefore)
 {
-  m_queue.update_job(sjob, suggested);
+  m_queue.update_job(sjob, suggested, reset_notbefore);
 }
 
+void OsdScrub::push_nb_out(
```
Comment (athanatos) [inline src/osd/scrubber/osd_scrub.cc:438]: Something like update_job_on_failure?

File: src/osd/scrubber/scrub_job.cc [hunk 2]
```diff
@@ -43,21 +43,40 @@ ostream& operator<<(ostream& out, const ScrubJob& sjob)
 }
 }  // namespace std
 
-void ScrubJob::update_schedule(const Scrub::scrub_schedule_t& adjusted)
+void ScrubJob::update_schedule(
+    const Scrub::scrub_schedule_t& adjusted,
+    bool reset_nb)
```
Comment (athanatos) [inline src/osd/scrubber/scrub_job.cc:48]: reset_failure_penalty maybe?

**Review by athanatos — APPROVED** [review]
lgtm other than naming nits

**Comment by ronen-fr** [comment]
Re an additional comment made in the review: should update_job() with reset=true be given a unique name,
and understood as initializing the target vs modifying it:
after going over the changes: this will be part of the next phase of this refactoring effort.

**Comment by ronen-fr** [comment]
jenkins test api

**Pull request merged**